File: CodePeople/src/kr/or/ddit/cmm/service/CheckRight.java
package kr.or.ddit.cmm.service;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

import kr.or.ddit.member.service.IMemberService;
import kr.or.ddit.member.service.MemberServiceImpl;
import kr.or.ddit.member.vo.MemberVO;

@WebServlet("/CheckRight.do")
public class CheckRight extends HttpServlet{

	@Override
	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
		
		HttpSession session = req.getSession();
        MemberVO loginMember = session.getAttribute("memVO") == null?
                null : (MemberVO)session.getAttribute("memVO");
        IMemberService memberService = MemberServiceImpl.getInstance();
        
        System.out.println(loginMember == null);
        
        req.setAttribute("result", 0);
        req.setAttribute("hrNo", 0);
        req.setAttribute("memNo", 0);
        req.setAttribute("comNo", 0);
        
        if(loginMember == null) {
            
            req.setAttribute("result", 0);
            
            req.getRequestDispatcher("/WEB-INF/jsp/CheckRight.jsp").forward(req, resp);
            
        }else {
            
            int memCode = loginMember.getMemCode();
            int memNo = loginMember.getMemNo();
            
            req.setAttribute("memNo", memNo);
            
            if(memCode == 2) {
            	
                int check = memberService.checkRegisterdCorp(memNo);
                if(check > 0) {
                    int comNo = memberService.getComNo(memNo);
                    req.setAttribute("comNo", comNo);
                	req.setAttribute("result", 2);
                	req.getRequestDispatcher("/WEB-INF/jsp/CheckRight.jsp").forward(req, resp);
                }else {
                	req.setAttribute("result", 3);
                	req.getRequestDispatcher("/WEB-INF/jsp/CheckRight.jsp").forward(req, resp);
                }
            }else if(memCode == 1) {
            	int hrNo = memberService.getHrNo(memNo);
            	req.setAttribute("hrNo", hrNo);
            	req.setAttribute("result", 1);
            	req.getRequestDispatcher("/WEB-INF/jsp/CheckRight.jsp").forward(req, resp);
            }
        }
	}
}

File: CodePeople/WebContent/js/CheckRight.js
// 세션에서 회원이 기업회원임과 등록한 기업이 있다는 것을 확인하여
// 등록한 기업의 기업번호를 리턴하는 메서드, 등록한 기업이 없다면 0을 반환
function getComNo(){
	
	$.ajax({
		url: '/CodePeople/CheckRight.do',
		type: 'get',
		dataType: 'json',
		success: function(res){
			console.log('회원구분코드 : ' + res.memCode + '\n' + '회원번호 : ' + res.memNo
					 + '인재번호 : ' + res.hrNo + '\n' + '기업번호 : ' + res.comNo);
			return res.comNo;
		},
		error: function(err){
			alert(`status : ${err.status}`);
		}		
	})
	
}

//세션에서 회원이 개인회원임을 확인하여
//인재번호를 리턴하는 메서드, 비회원이거나 기업회원인 경우 0을 반환
function getHrNo(){
	
	$.ajax({
		url: '/CodePeople/CheckRight.do',
		type: 'get',
		dataType: 'json',
		success: function(res){
			
			console.log('회원구분코드 : ' + res.memCode + '\n' + '회원번호 : ' + res.memNo
					 + '인재번호 : ' + res.hrNo + '\n' + '기업번호 : ' + res.comNo);
			return res.hrNo;
		},
		error: function(err){
			alert(`status : ${err.status}`);
		}		
	})
	
}

//세션에서 회원의 회원번호를 반환하는 메서드, 비회원인 경우 0을 반환
function getMemNo(){
	
	$.ajax({
		url: '/CodePeople/CheckRight.do',
		type: 'get',
		dataType: 'json',
		success: function(res){
			
			console.log('회원구분코드 : ' + res.memCode + '\n' + '회원번호 : ' + res.memNo
					 + '인재번호 : ' + res.hrNo + '\n' + '기업번호 : ' + res.comNo);
			return res.memNo;
		},
		error: function(err){
			alert(`status : ${err.status}`);
		}		
	})
	
}

function getMemCodeNo(){
	
	$.ajax({
		url: '/CodePeople/CheckRight.do',
		type: 'get',
		dataType: 'json',
		success: function(res){
			
			console.log('회원구분코드 : ' + res.memCode + '\n' + '회원번호 : ' + res.memNo
					+ '인재번호 : ' + res.hrNo + '\n' + '기업번호 : ' + res.comNo);
			return res.memCode;
		},
		error: function(err){
			alert(`status : ${err.status}`);
		}		
	})
	
}